### PR TITLE
UserIdentity without Migration

### PR DESCRIPTION
**Pull request by Maxiboy441**

### CODE REVIEW

**Comment by Maxiboy441** [comment]
After you put something wrong in the register page, the view changes. Please make sure it allways follows the concept

image
image

**Comment by Maxiboy441** [comment]
image

After registering this comes up. Please remove the whole register conformation logic or implement it (would do the first because it can be hard todo and is not needed for our usecase)

**Comment by Maxiboy441** [comment]
image

Username wants email??? Please decide if you login with email or username (best case the unique value)

**Comment by Maxiboy441** [comment]
image
image

Size of the container seems to change if there is a wrong value. Please fix this

**Comment by Maxiboy441** [comment]
image

To many values, specially the ones which are duplcates. What is the diffrent between Name and UserName for example?

Is the UserId supposed to be a varChar? In the concept its a INT


image

Why is there a username which is my email when there is a name?



Please make sure its like the concept. Thx


P.S: One thing you did, which is much better than in the concept, is that you only store the password hash. If this works its much better than storing the clear password and I would like to encourage you keeping the password hash instead of the cleartext password of the concept (if it works)

**Comment by Maxiboy441** [comment]
image

Container gets longer if you put in wrong credentials, this souldnt happen

**Comment by Maxiboy441** [comment]
After login:
image



Maybe following contains the fix:

https://stackoverflow.com/questions/46761397/invalidoperationexception-cannot-create-a-dbset-for-role-because-this-type-is

File: Project/Project/Areas/Identity/Pages/Account/Login.cshtml
Comment (Maxiboy441) [inline Project/Project/Areas/Identity/Pages/Account/Login.cshtml:1]: Please use bootstrap wherever possible insted of plan css styles

File: Project/Project/Areas/Identity/Pages/Account/Login.cshtml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            <label asp-for="Input.Password" class="col-md-4 col-form-label">Password</label>
+            
+            <div class="mb-3 align-items-left">
+                <input asp-for="Input.Password" class="form-control w-100 rounded-pill mb-2" autocomplete="password" aria-required="true" placeholder="Password"/>
+                <span asp-validation-for="Input.Password" class="text-danger"></span>
+            </div>
+
+            <div class="mb-4 mt-4">
+                <button id="login-submit" type="submit" class="w-100 btn fw-bold" style="background: #7432F9">Login</button>
+            </div>
```
Comment (Maxiboy441) [inline Project/Project/Areas/Identity/Pages/Account/Login.cshtml:31]: Check if you can use the PrimaryButton component here

File: Project/Project/Areas/Identity/Pages/Account/Login.cshtml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            
+            <div class="d-flex flex-column justify-content-center align-items-center" style="margin-left: 30px">
+                <hr class="w-75 text-light"/>
+                <a class="w-75 btn btn-primary btn-success fw-bold" asp-page="./Register" asp-route-returnUrl="@Model.ReturnUrl">Create new account</a>
```
Comment (Maxiboy441) [inline Project/Project/Areas/Identity/Pages/Account/Login.cshtml:35]: Check if you can use the PrimaryButton component here

File: Project/Project/Areas/Identity/Pages/Account/Register.cshtml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            </div>
+
+            <div class="mb-3">
+                <label asp-for="Input.ConfirmPassword" class="col-form-label">Confirm Password</label>
+                <input asp-for="Input.ConfirmPassword" class="form-control w-100 rounded-pill mb-2" autocomplete="password" aria-required="true" placeholder="Confirm Password"/>
+                <span asp-validation-for="Input.ConfirmPassword" class="text-danger"></span>
+            </div>
+
+            <div class="mb-4 mt-4">
+                <button id="registerSubmit" type="submit" class="w-100 btn fw-bold" style="background: #7432F9">Sign Up</button>
```
Comment (Maxiboy441) [inline Project/Project/Areas/Identity/Pages/Account/Register.cshtml:39]: Check if you can use the PrimaryButton component here

File: Project/Project/Migrations/20240630205010_create_users_table_migration.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                    LockoutEnd = table.Column<DateTimeOffset>(type: "datetime(6)", nullable: true),
+                    LockoutEnabled = table.Column<bool>(type: "tinyint(1)", nullable: false),
+                    AccessFailedCount = table.Column<int>(type: "int", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_User", x => x.Id);
+                })
+                .Annotation("MySql:CharSet", "utf8mb4");
+        }
```
Comment (Maxiboy441) [inline Project/Project/Migrations/20240630205010_create_users_table_migration.cs:54]: Like said in other comments, please adjusr according to the concept

File: Project/Project/Migrations/DataContextModelSnapshot.cs [hunk 5]
```diff
@@ -18,6 +19,68 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+
+                    b.Property<bool>("TwoFactorEnabled")
+                        .HasColumnType("tinyint(1)");
+
+                    b.Property<string>("UserName")
+                        .HasColumnType("longtext");
+
+                    b.HasKey("Id");
+
+                    b.ToTable("User");
```
Comment (Maxiboy441) [inline Project/Project/Migrations/DataContextModelSnapshot.cs:81]: Like said in other comments, please adjusr according to the concept

File: Project/Project/Migrations/20240630205010_create_users_table_migration.cs (hunk 4 above)
Comment (mehrb98) [reply Project/Project/Migrations/20240630205010_create_users_table_migration.cs:54]: OK

**Pull request closed without merging**